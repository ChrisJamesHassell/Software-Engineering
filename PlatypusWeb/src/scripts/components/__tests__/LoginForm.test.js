import React from 'react';
import { shallow, mount, render } from 'enzyme';
import LoginForm from '../Forms/LoginForm.js';
import TextInput from '../Forms/TextInput';
import {regex} from '../Forms/LoginForm.js';
import { textInput } from '../dataFixtures/fixtures';

const props = { textInput };


// describe what we are testing
describe('LoginForm', () => {
  const loginForm = shallow(<LoginForm  {...props}/>);
 
  // make our assertion and what we expect to happen 
  it('should render a local LoginForm', () => {
    // console.log(loginForm.debug());
    expect(loginForm.find('form').exists()).toBe(true)
  });

  // it('should render username stuff', () => {
  //   expect(loginForm.find('FormGroup').at(0).props().children).toEqual('User Name');
  // });


  it('should say Enter user name', () =>{
    expect (loginForm.find('TextInput').exists()).toBe(true);
  });
})

  // it('should say Enter user name', () =>{
  //   expect (loginForm.find('TextInput').at(0).props().children).toEqual('userName');
  // });

//   describe('and updating the username', () => {
//     beforeEach(() => {
//         loginForm.find('TextInput').at(0).simulate('change', { target: {value: 'change username'} })
//     });

//     it('updates the username', () => {
//         // console.log(stackForm.state());
//         expect(loginForm.state().title).toEqual('change title');
//     });
// });
// });


  // it('renders a password input', () => {
  //   expect(shallow(<TextInput />).find('#User Name').exists()).toBe(true)
  // })

// })
// test('there should not be all lower', () => {
//   expect ('teamI').not.toMatch(/I/);
// });

// describe ('<TextInput />', () => {
//   it('userName accepts id prop', () => {
//     const wrapper = shallow(<TextInput )
  