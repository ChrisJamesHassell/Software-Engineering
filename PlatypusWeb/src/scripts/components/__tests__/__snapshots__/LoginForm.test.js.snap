// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    textInput={
      Object {
        "state": Object {
          "requiresValidation": false,
          "value": "testVal",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "http://localhost:8080/api/user/login",
      "children": Array [
        <FormGroup
          bsClass="form-group"
          controlId="username"
        >
          <InputGroup
            bsClass="input-group"
          >
            <InputGroupAddon
              bsClass="input-group-addon"
            >
              <Glyphicon
                bsClass="glyphicon"
                glyph="user"
              />
            </InputGroupAddon>
            <TextInput
              label="User Name"
              maxLength="32"
              name="username"
              placeholder="Enter your user name"
              required={true}
              type="text"
              updateVals={[Function]}
            />
          </InputGroup>
        </FormGroup>,
        <FormGroup
          bsClass="form-group"
          controlId="pass"
        >
          <InputGroup
            bsClass="input-group"
          >
            <InputGroupAddon
              bsClass="input-group-addon"
            >
              <Glyphicon
                bsClass="glyphicon"
                glyph="lock"
              />
            </InputGroupAddon>
            <TextInput
              label="Password"
              maxLength="32"
              name="password"
              placeholder="Enter your password"
              required={true}
              type="password"
              updateVals={[Function]}
            />
          </InputGroup>
        </FormGroup>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="success"
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="button"
        >
          Login
        </Button>,
      ],
      "encType": "multipart/form-data",
      "method": "POST",
      "onKeyUp": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <InputGroup
            bsClass="input-group"
          >
            <InputGroupAddon
              bsClass="input-group-addon"
            >
              <Glyphicon
                bsClass="glyphicon"
                glyph="user"
              />
            </InputGroupAddon>
            <TextInput
              label="User Name"
              maxLength="32"
              name="username"
              placeholder="Enter your user name"
              required={true}
              type="text"
              updateVals={[Function]}
            />
          </InputGroup>,
          "controlId": "username",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "input-group",
            "children": Array [
              <InputGroupAddon
                bsClass="input-group-addon"
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="user"
                />
              </InputGroupAddon>,
              <TextInput
                label="User Name"
                maxLength="32"
                name="username"
                placeholder="Enter your user name"
                required={true}
                type="text"
                updateVals={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "input-group-addon",
                "children": <Glyphicon
                  bsClass="glyphicon"
                  glyph="user"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "glyphicon",
                  "glyph": "user",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "User Name",
                "maxLength": "32",
                "name": "username",
                "placeholder": "Enter your user name",
                "required": true,
                "type": "text",
                "updateVals": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <InputGroup
            bsClass="input-group"
          >
            <InputGroupAddon
              bsClass="input-group-addon"
            >
              <Glyphicon
                bsClass="glyphicon"
                glyph="lock"
              />
            </InputGroupAddon>
            <TextInput
              label="Password"
              maxLength="32"
              name="password"
              placeholder="Enter your password"
              required={true}
              type="password"
              updateVals={[Function]}
            />
          </InputGroup>,
          "controlId": "pass",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "input-group",
            "children": Array [
              <InputGroupAddon
                bsClass="input-group-addon"
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="lock"
                />
              </InputGroupAddon>,
              <TextInput
                label="Password"
                maxLength="32"
                name="password"
                placeholder="Enter your password"
                required={true}
                type="password"
                updateVals={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "input-group-addon",
                "children": <Glyphicon
                  bsClass="glyphicon"
                  glyph="lock"
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "glyphicon",
                  "glyph": "lock",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Password",
                "maxLength": "32",
                "name": "password",
                "placeholder": "Enter your password",
                "required": true,
                "type": "password",
                "updateVals": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "success",
          "children": "Login",
          "disabled": true,
          "onClick": [Function],
          "style": Object {
            "width": "100%",
          },
          "type": "button",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "http://localhost:8080/api/user/login",
        "children": Array [
          <FormGroup
            bsClass="form-group"
            controlId="username"
          >
            <InputGroup
              bsClass="input-group"
            >
              <InputGroupAddon
                bsClass="input-group-addon"
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="user"
                />
              </InputGroupAddon>
              <TextInput
                label="User Name"
                maxLength="32"
                name="username"
                placeholder="Enter your user name"
                required={true}
                type="text"
                updateVals={[Function]}
              />
            </InputGroup>
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            controlId="pass"
          >
            <InputGroup
              bsClass="input-group"
            >
              <InputGroupAddon
                bsClass="input-group-addon"
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="lock"
                />
              </InputGroupAddon>
              <TextInput
                label="Password"
                maxLength="32"
                name="password"
                placeholder="Enter your password"
                required={true}
                type="password"
                updateVals={[Function]}
              />
            </InputGroup>
          </FormGroup>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="success"
            disabled={true}
            onClick={[Function]}
            style={
              Object {
                "width": "100%",
              }
            }
            type="button"
          >
            Login
          </Button>,
        ],
        "encType": "multipart/form-data",
        "method": "POST",
        "onKeyUp": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <InputGroup
              bsClass="input-group"
            >
              <InputGroupAddon
                bsClass="input-group-addon"
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="user"
                />
              </InputGroupAddon>
              <TextInput
                label="User Name"
                maxLength="32"
                name="username"
                placeholder="Enter your user name"
                required={true}
                type="text"
                updateVals={[Function]}
              />
            </InputGroup>,
            "controlId": "username",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "input-group",
              "children": Array [
                <InputGroupAddon
                  bsClass="input-group-addon"
                >
                  <Glyphicon
                    bsClass="glyphicon"
                    glyph="user"
                  />
                </InputGroupAddon>,
                <TextInput
                  label="User Name"
                  maxLength="32"
                  name="username"
                  placeholder="Enter your user name"
                  required={true}
                  type="text"
                  updateVals={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "input-group-addon",
                  "children": <Glyphicon
                    bsClass="glyphicon"
                    glyph="user"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "glyphicon",
                    "glyph": "user",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "User Name",
                  "maxLength": "32",
                  "name": "username",
                  "placeholder": "Enter your user name",
                  "required": true,
                  "type": "text",
                  "updateVals": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <InputGroup
              bsClass="input-group"
            >
              <InputGroupAddon
                bsClass="input-group-addon"
              >
                <Glyphicon
                  bsClass="glyphicon"
                  glyph="lock"
                />
              </InputGroupAddon>
              <TextInput
                label="Password"
                maxLength="32"
                name="password"
                placeholder="Enter your password"
                required={true}
                type="password"
                updateVals={[Function]}
              />
            </InputGroup>,
            "controlId": "pass",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "input-group",
              "children": Array [
                <InputGroupAddon
                  bsClass="input-group-addon"
                >
                  <Glyphicon
                    bsClass="glyphicon"
                    glyph="lock"
                  />
                </InputGroupAddon>,
                <TextInput
                  label="Password"
                  maxLength="32"
                  name="password"
                  placeholder="Enter your password"
                  required={true}
                  type="password"
                  updateVals={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "input-group-addon",
                  "children": <Glyphicon
                    bsClass="glyphicon"
                    glyph="lock"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "glyphicon",
                    "glyph": "lock",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Password",
                  "maxLength": "32",
                  "name": "password",
                  "placeholder": "Enter your password",
                  "required": true,
                  "type": "password",
                  "updateVals": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "success",
            "children": "Login",
            "disabled": true,
            "onClick": [Function],
            "style": Object {
              "width": "100%",
            },
            "type": "button",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
